### PR TITLE
feat: Adding Enter Claims Evaluation Data UI #42

### PR DESCRIPTION
**Pull request by brian-seruwuge**
**IN REVIEW**
add form to enable users add agents, type of claim etc

### CODE REVIEW

File: src/views/components/EnterClaimsEvaluationData/index.tsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+class EnterClaimsEvaluationData extends React.Component<
+  EnterClaimsEvaluationDataProps,
+  EnterClaimsEvaluationDataState
+> {
+  constructor(props) {
+    super(props);
+    this.state = {
+      Claims: {
+        claimType: "",
+        workflowNumber: 0,
```
Comment (epicallan) [inline src/views/components/EnterClaimsEvaluationData/index.tsx]: should be empty i.e null intially

File: src/views/components/EnterClaimsEvaluationData/index.tsx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+            placeholder="Last Name"
+            value={this.state.Claims.workflowNumber}
+            type="number"
+            onChange={this.handleInput}
+            required={true}
+          />
+        </Form.Field>
+        <Form>
+          <Form.Field>
+            <label>comment(optional)</label>
```
Comment (epicallan) [inline src/views/components/EnterClaimsEvaluationData/index.tsx]: You don't have to add optional in brackets. It was just an indicator that the field is optional

File: src/views/components/EnterClaimsEvaluationData/index.tsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+            options={this.props.claimTypes}
+            onChange={this.handleInput}
+          />
+        </Form.Field>
+        <Form.Field>
+          <label>All parameters met </label>
+          <Checkbox label={this.state.Claims.allParametersMet} />
+        </Form.Field>
+        <Form.Field>
+          <label>work flow number(required)</label>
```
Comment (epicallan) [inline src/views/components/EnterClaimsEvaluationData/index.tsx]: No need to add required in brackets. It was just a guideline

File: src/views/components/EnterClaimsEvaluationData/index.tsx [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+            fluid={true}
+            search={true}
+            selection={true}
+            options={this.props.claimTypes}
+            onChange={this.handleInput}
+          />
+        </Form.Field>
+        <Form.Field>
+          <label>All parameters met </label>
+          <Checkbox label={this.state.Claims.allParametersMet} />
```
Comment (epicallan) [inline src/views/components/EnterClaimsEvaluationData/index.tsx]: Chequebook is missing a handler.

File: src/views/components/EnterClaimsEvaluationData/index.tsx [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+  EnterClaimsEvaluationDataState
+> {
+  constructor(props) {
+    super(props);
+    this.state = {
+      Claims: {
+        claimType: "",
+        workflowNumber: 0,
+        agent: "",
+        allParametersMet: true,
```
Comment (epicallan) [inline src/views/components/EnterClaimsEvaluationData/index.tsx]: The default state should be `false`.

File: src/views/components/EnterClaimsEvaluationData/index.tsx [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+        comments: ""
+      }
+    };
+
+    this.handleInput = this.handleInput.bind(this);
```
Comment (epicallan) [inline src/views/components/EnterClaimsEvaluationData/index.tsx]: no need to use bind if you use arrow functions

File: src/views/components/EnterClaimsEvaluationData/index.tsx [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+  }
+
+  public submitForm = async (): Promise<void> => {
+    return this.props.onSubmit(this.state.Claims);
+  };
+
+  public handleInput(event: any): void {
```
Comment (epicallan) [inline src/views/components/EnterClaimsEvaluationData/index.tsx]: use arrow function to get rid of `bind`

**Comment by epicallan** [comment]
You should also submit current `evaluator` in the `onSubmit` function. Take a look at updated data types https://github.com/freelyformd/eval-cso/issues/42#issue-427534783

File: src/views/components/EnterClaimsEvaluationData/index.tsx [hunk 8]
```diff
@@ -0,0 +1,108 @@
+import * as React from "react";
+import { Checkbox, Dropdown, Form, TextArea } from "semantic-ui-react";
+
+export interface EnterClaimsEvaluationDataProps {
+  agents: any; // usernames
```
Comment (lubegasimon) [inline src/views/components/EnterClaimsEvaluationData/index.tsx]: https://github.com/freelyformd/eval-cso/issues/29#issuecomment-478338463 ... so to put it in a format that `options` props in `Dropdown` accepts, you'll have to do something of this kind;
```
  public agentNameOptions = (agentNames: string[]): AgentNameOptions[] => {
    return agentNames.map(
      (agentName): AgentNameOptions => {
        return {
          key: agentName,
          value: agentName.toLowerCase(),
          text: agentName.toUpperCase()
        };
      }
    );
  };
```

Comment (epicallan) [reply src/views/components/EnterClaimsEvaluationData/index.tsx]: Don't use `any` type for things that can be properly typed.

File: src/views/components/EnterClaimsEvaluationData/index.tsx (hunk 6 above)
Comment (brian-seruwuge) [reply src/views/components/EnterClaimsEvaluationData/index.tsx]: handleInput is not an arrow function, that's why bind is used
its only onSubmit that is an arrow function
cant you mix them up

**Comment by brian-seruwuge** [comment]
> You should also submit current `evaluator` in the `onSubmit` function. Take a look at updated data types [#42 (comment)](https://github.com/freelyformd/eval-cso/issues/42#issue-427534783)

File: src/views/components/EnterClaimsEvaluationData/index.tsx (hunk 3 above)
Comment (brian-seruwuge) [reply src/views/components/EnterClaimsEvaluationData/index.tsx]: those were just reminder i had put just for me to remember that i had to make the field required, though I committed with them. 
ill remove them

File: src/views/components/EnterClaimsEvaluationData/index.tsx (hunk 6 above)
Comment (epicallan) [reply src/views/components/EnterClaimsEvaluationData/index.tsx]: We want to avoid using bind. We want to be consistent please use arrow functions to avoid using bind.

File: src/views/components/EnterClaimsEvaluationData/index.tsx [hunk 9]
```diff
@@ -1,23 +1,34 @@
...
   claimTypes: any;
+  evaluator: string;
   onSubmit: (data: ClaimsEvaluation) => Promise<void>;
   loading: boolean;
 }
 
 interface ClaimsEvaluation {
   claimType: string;
-  workflowNumber: number;
+  workflowNumber: any;
```
Comment (lubegasimon) [inline src/views/components/EnterClaimsEvaluationData/index.tsx]: use `number` type

File: src/views/components/EnterClaimsEvaluationData/index.tsx [hunk 10]
```diff
@@ -26,34 +37,45 @@ class EnterClaimsEvaluationData extends React.Component<
 > {
   constructor(props) {
     super(props);
+
     this.state = {
       Claims: {
         claimType: "",
-        workflowNumber: 0,
+        workflowNumber: "",
```
Comment (lubegasimon) [inline src/views/components/EnterClaimsEvaluationData/index.tsx]: use `null` as initial `state` value

File: src/views/components/EnterClaimsEvaluationData/index.tsx [hunk 11]
```diff
@@ -1,23 +1,34 @@
...
 export interface EnterClaimsEvaluationDataProps {
   agents: any; // usernames
   claimTypes: any;
+  evaluator: string;
   onSubmit: (data: ClaimsEvaluation) => Promise<void>;
   loading: boolean;
 }
 
 interface ClaimsEvaluation {
   claimType: string;
```
Comment (lubegasimon) [inline src/views/components/EnterClaimsEvaluationData/index.tsx]: in your storybook, this is of type `array` (`[]`), but here, it's different...

File: src/views/components/EnterClaimsEvaluationData/index.tsx [hunk 12]
```diff
@@ -0,0 +1,113 @@
+import * as React from "react";
+import { Button, Checkbox, Dropdown, Form, TextArea } from "semantic-ui-react";
+
+export interface EnterClaimsEvaluationDataProps {
```
Comment (epicallan) [inline src/views/components/EnterClaimsEvaluationData/index.tsx]: These prop type names are un-necessary long let's stick to just `Props`

File: src/views/components/EnterClaimsEvaluationData/index.tsx [hunk 13]
```diff
@@ -0,0 +1,113 @@
...
+  claimType: string;
+  workflowNumber: number;
+  agent: string;
+  allParametersMet: boolean;
+  comments: string;
+  evaluator: string;
+  loading: boolean;
+}
+
+export interface EnterClaimsEvaluationDataState {
```
Comment (epicallan) [inline src/views/components/EnterClaimsEvaluationData/index.tsx]: Just stick to naming this as `State`

File: src/views/components/EnterClaimsEvaluationData/index.tsx [hunk 14]
```diff
@@ -0,0 +1,113 @@
...
+  Claims: ClaimsEvaluation;
```
Comment (epicallan) [inline src/views/components/EnterClaimsEvaluationData/index.tsx]: I don't think there is any benefit in having all the component state under the `claims` field.  That would have been ok if the state was tracking different entities. So the component state should  just be the type you have for `ClaimEvaluation`

Comment (epicallan) [reply src/views/components/EnterClaimsEvaluationData/index.tsx]: Object or type field names are supposed to be lower case not capitalised as you did for `Claims: ClaimsEvaluation`

File: src/views/components/EnterClaimsEvaluationData/index.tsx [hunk 15]
```diff
@@ -0,0 +1,113 @@
...
+}
+
+interface ClaimsEvaluation {
+  claimType: string;
+  workflowNumber: number;
+  agent: string;
+  allParametersMet: boolean;
+  comments: string;
+  evaluator: string;
+  loading: boolean;
```
Comment (epicallan) [inline src/views/components/EnterClaimsEvaluationData/index.tsx]: This doesn't need to be in `state` you are not evening using it. Why add it there?

File: src/views/components/EnterClaimsEvaluationData/index.tsx [hunk 16]
```diff
@@ -0,0 +1,113 @@
...
+    const value = event.target.value;
+    const name = event.target.name;
+    const changedClaim = { [value]: { value, name } };
+    const Claims = { ...this.state.Claims, ...changedClaim };
+    this.setState({ Claims });
+  };
+
+  public render() {
+    return (
+      <Form onSubmit={this.submitForm}>
```
Comment (epicallan) [inline src/views/components/EnterClaimsEvaluationData/index.tsx]: Form is missing the loading attribute whose value is from `this.props.loading`

File: src/views/components/EnterClaimsEvaluationData/index.tsx (hunk 8 above)
Comment (epicallan) [reply src/views/components/EnterClaimsEvaluationData/index.tsx]: The documentation for [Dropdown](https://react.semantic-ui.com/modules/dropdown/#types-search-selection)  clearly shows what options data is supposed to look like, why have you set it to `any` ??????????!!!

File: src/views/components/EnterClaimsEvaluationData/index.tsx (hunk 11 above)
Comment (epicallan) [reply src/views/components/EnterClaimsEvaluationData/index.tsx]: This is for state, it is correctly a string

File: src/views/components/EnterClaimsEvaluationData/index.tsx [hunk 17]
```diff
@@ -0,0 +1,113 @@
+import * as React from "react";
+import { Button, Checkbox, Dropdown, Form, TextArea } from "semantic-ui-react";
+
+export interface EnterClaimsEvaluationDataProps {
+  agents: any; // usernames
+  claimTypes: any;
```
Comment (epicallan) [inline src/views/components/EnterClaimsEvaluationData/index.tsx]: still just using `any` 😢

File: src/views/components/EnterClaimsEvaluationData/index.stories.tsx [hunk 18]
```diff
@@ -0,0 +1,15 @@
+import { action } from "@storybook/addon-actions";
+import { storiesOf } from "@storybook/react";
+import React from "react";
+import EnterClaimsEvaluationData from "./index";
+
+const props = {
+  loading: false,
+  agents: [], // usernames
```
Comment (epicallan) [inline src/views/components/EnterClaimsEvaluationData/index.stories.tsx]: You should add agents username after addressing comments on this PR such that we can test out the component properly. You should also do the same for `claimTypes`

File: src/views/components/EnterClaimsEvaluationData/index.tsx [hunk 19]
```diff
@@ -0,0 +1,113 @@
...
+            fluid={true}
+            search={true}
+            selection={true}
+            options={this.props.agents}
+            onChange={this.handleInput}
+          />
+        </Form.Field>
+        <Form.Field>
+          <label>Type of claim</label>
+          <Dropdown
```
Comment (epicallan) [inline src/views/components/EnterClaimsEvaluationData/index.tsx]: missing `required` attribute

File: src/views/components/EnterClaimsEvaluationData/index.tsx [hunk 20]
```diff
@@ -0,0 +1,113 @@
...
+
+class EnterClaimsEvaluationData extends React.Component<
+  EnterClaimsEvaluationDataProps,
+  EnterClaimsEvaluationDataState
+> {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      Claims: {
```
Comment (epicallan) [inline src/views/components/EnterClaimsEvaluationData/index.tsx]: Fields that are optional can't be just an empty string reason being is that when they get submitted we will have empty strings in the database instead of a `null` value.

File: src/views/components/EnterClaimsEvaluationData/index.tsx (hunk 10 above)
Comment (epicallan) [reply src/views/components/EnterClaimsEvaluationData/index.tsx]: Actually using `null` as an initial prop for inputs generates error warning. So we need default values and then possibly turn them to null before submission, for-instance workflowNumber.

File: src/views/components/EnterClaimsEvaluationData/index.tsx (hunk 20 above)
Comment (brian-seruwuge) [reply src/views/components/EnterClaimsEvaluationData/index.tsx]: empty strings conflict with the 'number' type that has to be assigned to workflowNumber giving an error

File: src/views/components/EnterClaimsEvaluationData/index.stories.tsx (hunk 18 above)
Comment (epicallan) [reply src/views/components/EnterClaimsEvaluationData/index.stories.tsx]: This was not addressed ??

File: src/views/components/EnterClaimsEvaluationData/index.tsx (hunk 19 above)
Comment (epicallan) [reply src/views/components/EnterClaimsEvaluationData/index.tsx]: `Dropdowns` can't have a required field. So you need to write some custom validation code to make sure that a claim is always set. see this component https://github.com/freelyformd/eval-cso/blob/master/src/views/components/EvaluationDataForm/index.tsx for an example

File: src/views/components/EnterClaimsEvaluationData/index.tsx [hunk 21]
```diff
@@ -0,0 +1,158 @@
...
+  public submitForm = async (): Promise<void> => {
+    return this.props.onSubmit(this.state.claims);
+  };
+
+  public handleDropDownInput = (_event, { name, value }): void => {
+    const claims = { ...this.state.claims, [name]: value };
+    return this.setState({ claims });
+  };
+
+  public AgentOptions = (agents: string[]): AgentOptions[] => {
```
Comment (epicallan) [inline src/views/components/EnterClaimsEvaluationData/index.tsx]: function names are not supposed to be capitalised

File: src/views/components/EnterClaimsEvaluationData/index.tsx [hunk 22]
```diff
@@ -0,0 +1,158 @@
...
+        return {
+          key: agent,
+          value: agent.toLowerCase(),
+          text: agent
+        };
+      }
+    );
+  };
+
+  public ClaimTypeOptions = (claimTypes: string[]): ClaimTypeOptions[] => {
```
Comment (epicallan) [inline src/views/components/EnterClaimsEvaluationData/index.tsx]: don't capitalise function names

File: src/views/components/EnterClaimsEvaluationData/index.tsx [hunk 23]
```diff
@@ -0,0 +1,158 @@
...
+          <Checkbox
+            label={this.state.claims.allParametersMet}
+            onChange={this.handleInput}
+          />
+        </Form.Field>
+        <Form.Field>
+          <label>work flow number</label>
+          <input
+            placeholder="Add workflow number"
+            value={this.state.claims.workflowNumber}
```
Comment (epicallan) [inline src/views/components/EnterClaimsEvaluationData/index.tsx]: you can by default set `this.state.claims.workflowNumber` to zero and then write something like.  which will make the input value to be empty as long as its zero
```suggestion
            value={this.state.claims.workflowNumber || ""}
```

File: src/views/components/EnterClaimsEvaluationData/index.tsx [hunk 24]
```diff
@@ -0,0 +1,158 @@
...
+}
+
+class EnterClaimsEvaluationData extends React.Component<Props, State> {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      claims: {
+        claimType: "",
+        workflowNumber: null,
```
Comment (epicallan) [inline src/views/components/EnterClaimsEvaluationData/index.tsx]: set this to zero. I left a comment on how it's handled on the workflow number input

Comment (brian-seruwuge) [reply src/views/components/EnterClaimsEvaluationData/index.tsx]: setting workflowNumber to zero and using 'value={this.state.claims.workflowNumber || ""}' makes the field not to display any value when any of the numbers is typed.
I found using setting 'workflowNumber' type to 'undefined' suitable. 
hope this does not violate any rule

File: src/views/components/EnterClaimsEvaluationData/index.tsx (hunk 23 above)
Comment (epicallan) [reply src/views/components/EnterClaimsEvaluationData/index.tsx]: This works fine after having set workflowNumber to zeron in; 
```this.state = {
      claims: {
        claimType: "",
        workflowNumber: 0,
        agent: "",
        allParametersMet: false,
        comments: "",
        evaluator: "",
        loading: this.props.loading
      },
      dropdownError: null
    };```

File: src/views/components/EnterClaimsEvaluationData/index.tsx [hunk 25]
```diff
@@ -0,0 +1,187 @@
...
+          text: claimType
+        };
+      }
+    );
+  };
+
+  public handleInput = (event): any => {
+    const value = event.target.value;
+    const name = event.target.name;
+    const claims = { ...this.state.claims, [value]: name };
```
Comment (epicallan) [inline src/views/components/EnterClaimsEvaluationData/index.tsx]:
```suggestion
    const claims = { ...this.state.claims, [name]: value};
```

File: src/views/components/EnterClaimsEvaluationData/index.tsx [hunk 26]
```diff
@@ -0,0 +1,187 @@
...
+            name="claim"
+          />
+          <Message
+            error={true}
+            header="claim input Error"
+            content={this.state.dropdownError}
+          />
+        </Form.Field>
+        <Form.Field>
+          <label>All parameters met </label>
```
Comment (epicallan) [inline src/views/components/EnterClaimsEvaluationData/index.tsx]:
```suggestion
            <Checkbox
            label= "All parameters met"
            onChange={this.handleInput}
          />
        </Form.Field>
```

File: src/views/components/EnterClaimsEvaluationData/index.tsx [hunk 27]
```diff
@@ -0,0 +1,187 @@
...
+    };
+  }
+
+  public submitForm = async (): Promise<void> => {
+    if (this.validate()) {
+      return this.props.onSubmit(this.state.claims);
+    }
+  };
+
+  public validate = (): boolean => {
```
Comment (epicallan) [inline src/views/components/EnterClaimsEvaluationData/index.tsx]: You are not handling validation for agent dropdown. `this.state.agent` can't be null on submission

File: src/views/components/EnterClaimsEvaluationData/index.tsx [hunk 28]
```diff
@@ -0,0 +1,187 @@
...
+        <Form.Field>
+          <label>Type of claim</label>
+          <Dropdown
+            placeholder="Select claim"
+            fluid={true}
+            search={true}
+            selection={true}
+            options={this.claimTypeOptions(this.props.claimTypes)}
+            onChange={this.handleDropDownInput}
+            required={true}
```
Comment (epicallan) [inline src/views/components/EnterClaimsEvaluationData/index.tsx]: has no effect, may aswell be removed

File: src/views/components/EnterClaimsEvaluationData/index.stories.tsx (hunk 18 above)
Comment (epicallan) [reply src/views/components/EnterClaimsEvaluationData/index.stories.tsx]: @brian-seruwuge  you need to add dummy data to your storybook so that you can test it out properly. PLEASE ADDRESS THIS!!

**Comment by brian-seruwuge** [comment]
story
log
Those are the screen shots

**Comment by brian-seruwuge** [comment]
log1
log screenshot

File: src/views/components/EnterClaimsEvaluationData/index.tsx [hunk 29]
```diff
@@ -0,0 +1,188 @@
...
+}
+
+interface ClaimsEvaluation {
+  claimType: string;
+  workflowNumber: number;
+  agent: string;
+  allParametersMet: boolean;
+  comments: string;
+  evaluator: string;
+  loading: boolean;
```
Comment (epicallan) [inline src/views/components/EnterClaimsEvaluationData/index.tsx]: loading shouldn't be part of `ClaimsEvaluation`. It could be part of the rest of the component state but not this.

File: src/views/components/EnterClaimsEvaluationData/index.tsx [hunk 30]
```diff
@@ -0,0 +1,188 @@
...
+  Button,
+  Checkbox,
+  Dropdown,
+  Form,
+  Message,
+  TextArea
+} from "semantic-ui-react";
+
+export interface Props {
+  agents: string[]; // usernames
```
Comment (epicallan) [inline src/views/components/EnterClaimsEvaluationData/index.tsx]: that comment is not required.

File: src/views/components/EnterClaimsEvaluationData/index.stories.tsx [hunk 31]
```diff
@@ -0,0 +1,15 @@
+import { action } from "@storybook/addon-actions";
+import { storiesOf } from "@storybook/react";
+import React from "react";
+import EnterClaimsEvaluationData from "./index";
+
+const props = {
+  loading: false,
+  agents: ["brian", "Alex", "Simon"], // usernames
```
Comment (epicallan) [inline src/views/components/EnterClaimsEvaluationData/index.stories.tsx]: comments should only be reserved for extremely hard to understand code.

**Comment by epicallan** [comment]
@brian-seruwuge from the log screen `evaluator` should never be an empty field, undefined and loading shouldn't exist either.

Your UI screenshot should capture the form with selected fields and the log screen can be as the one you have provided.

File: src/views/components/EnterClaimsEvaluationData/index.tsx [hunk 32]
```diff
@@ -0,0 +1,188 @@
...
+  claimTypes: string[];
+  evaluator: string;
```
Comment (epicallan) [inline src/views/components/EnterClaimsEvaluationData/index.tsx]: evaluator value never changes and thus shouldn't be part of the component state.

File: src/views/components/EnterClaimsEvaluationData/index.tsx [hunk 33]
```diff
@@ -0,0 +1,188 @@
...
+  loading: boolean;
+}
+
+interface ClaimsEvaluation {
+  claimType: string;
+  workflowNumber: number;
+  agent: string;
+  allParametersMet: boolean;
+  comments: string;
+  evaluator: string;
```
Comment (epicallan) [inline src/views/components/EnterClaimsEvaluationData/index.tsx]: evaluator value never changes and thus shouldn't be part of the component state.

File: src/views/components/EnterClaimsEvaluationData/index.tsx [hunk 34]
```diff
@@ -0,0 +1,188 @@
...
+    super(props);
+
+    this.state = {
+      claims: {
+        claimType: "",
+        workflowNumber: 0,
+        agent: "",
+        allParametersMet: false,
+        comments: "",
+        evaluator: "",
```
Comment (epicallan) [inline src/views/components/EnterClaimsEvaluationData/index.tsx]: remove evaluator shouldn't be part of state.

**Comment by brian-seruwuge** [comment]
> @brian-seruwuge from the log screen `evaluator` should never be an empty field, undefined and loading shouldn't exist either.

on Investigating i found out that if the handler on the check box is removed, we do not get 'undefined'
I left the handler in the code but **'undefined' does not exist in the final submitted object after clicking submit** though its **present by default in the object presented before submit is clicked**. check log

**Comment by epicallan** [comment]
@brian-seruwuge in your console log screenshot, `allParametersMet` value is false despite being checked and set to true in the UI. That shouldn't be the case.

File: src/views/components/EnterClaimsEvaluationData/index.tsx [hunk 35]
```diff
@@ -0,0 +1,184 @@
...
+}
+
+interface ClaimTypeOptions {
+  key: string;
+  value: string;
+  text: string;
+}
+
+export interface State {
+  claims: ClaimsEvaluation;
```
Comment (epicallan) [inline src/views/components/EnterClaimsEvaluationData/index.tsx]: rename to `claim` instead of `claims`

File: src/views/components/EnterClaimsEvaluationData/index.tsx [hunk 36]
```diff
@@ -0,0 +1,184 @@
...
+            fluid={true}
+            search={true}
+            selection={true}
+            options={this.claimTypeOptions(this.props.claimTypes)}
+            onChange={this.handleDropDownInput}
+            name="claimType"
+          />
+        </Form.Field>
+        <Form.Field>
+          <Checkbox label="All parameters met" onChange={this.handleInput} />
```
Comment (epicallan) [inline src/views/components/EnterClaimsEvaluationData/index.tsx]: `this.handleInput` can't be the handler used for `Checkbox` you need a different handler function.  See the `onChange` prop type signature for checkbox https://react.semantic-ui.com/modules/checkbox/

**Comment by brian-seruwuge** [comment]
sub1

**Comment by brian-seruwuge** [comment]
sub

File: src/views/components/ClaimEvaluation/index.tsx [hunk 37]
```diff
@@ -37,39 +36,43 @@ interface ClaimTypeOptions {
...
     };
   }
+  public toggle = () => {
+    this.setState({ allParametersMet: !this.state.allParametersMet });
+  };
 
   public submitForm = async (): Promise<void> => {
     if (this.validate()) {
-      return this.props.onSubmit(this.state.claims);
+      return this.props.onSubmit(this.state);
```
Comment (epicallan) [inline src/views/components/ClaimEvaluation/index.tsx]: Please don't submit the entire state to the `onSubmit` function. Functions should only receive the data they need to operate on. The `onSubmit` function basically calls a server API endpoint that adds data to a database and all it requires is `this.state.claim`

File: src/views/components/ClaimEvaluation/index.tsx [hunk 38]
```diff
@@ -37,39 +36,43 @@ interface ClaimTypeOptions {
...
+      claim: {
         claimType: "",
         workflowNumber: 0,
         agent: "",
-        allParametersMet: false,
         comments: ""
       },
-      dropdownError: null
+      dropdownError: null,
+      allParametersMet: false
```
Comment (epicallan) [inline src/views/components/ClaimEvaluation/index.tsx]: this should be part of `state.claim`. Why did you remove it  ?

File: src/views/components/ClaimEvaluation/index.tsx [hunk 39]
```diff
@@ -37,39 +36,43 @@ interface ClaimTypeOptions {
 }
 
 export interface State {
-  claims: ClaimsEvaluation;
+  claim: ClaimsEvaluation;
   dropdownError: string;
+  allParametersMet: boolean;
```
Comment (epicallan) [inline src/views/components/ClaimEvaluation/index.tsx]: why not leave under `claimsEvaluation` ?

File: src/views/components/ClaimEvaluation/index.tsx [hunk 40]
```diff
@@ -146,13 +149,18 @@ class EnterClaimsEvaluationData extends React.Component<Props, State> {
           />
         </Form.Field>
         <Form.Field>
-          <Checkbox label="All parameters met" onChange={this.handleInput} />
+          <Checkbox
```
Comment (epicallan) [inline src/views/components/ClaimEvaluation/index.tsx]: You are missing a `name` attribute on `Checkbox`. The name you provide to `Checkbox` is what is passed in ` public handleDropDownInput = (_event, { *name*, value })...`. So `name` should be  `allParametersMet`.

File: src/views/components/ClaimEvaluation/index.tsx [hunk 41]
```diff
@@ -50,19 +50,19 @@ class ClaimEvaluation extends React.Component<Props, State> {
...
-        comments: ""
+        comments: "",
+        allParametersMet: false
       },
-      dropdownError: null,
-      allParametersMet: false
+      dropdownError: null
     };
   }
   public toggle = () => {
```
Comment (epicallan) [inline src/views/components/ClaimEvaluation/index.tsx]: this function is never called, its thus pointless
